Refactor(pyavd,eos_designs)!: Use the same structured_config across all eos_designs modules (#5981)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

| `underlay_mutlticast` | `underlay_mutlticast_pim_sm` or `<node_type_key>.defaults.underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].nodes[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.nodes[].underlay_multicast.pim_sm.enabled` |
| `underlay_ospf_authentication.message_digest_keys[].key` | `underlay_ospf_authentication.message_digest_keys[].cleartext_key` |

### Enhanced conflict detection in `eos_designs`

In AVD 5.x, the conflict detection of AVD was done per data model (network_services, connected_enpoints, ...).
This implies that it was possible to declare conflicting information between data models (e.g. overwrite MLAG interface in connected_endpoints) which could lead to difficult to track errors.

Starting with AVD 6.0.0, conflict detection is now done across all data models allowing for a much stronger detection of these conflicting requirements.

> You may experience errors when migrating, but in general, these errors indicate that you had conflicting inputs across different configuration sections in your variables and you should adjust them accordingly.

Reviewer: ```suggestion
In AVD 5.x, conflict detection was done per data model (e.g., `network_services`, `connected_endpoints`, ...).
This implies that it was possible to declare conflicting information between data models (e.g., overwriting an MLAG interface in `connected_endpoints`), which could lead to difficult-to-track errors.

Starting with AVD 6.0.0, conflict detection is now done across all data models, allowing for much stronger detection of these conflicting requirements.

You may experience errors when migrating. In general, these errors indicate that you had conflicting inputs across different data models in your variables and you should adjust them accordingly.
```

| `underlay_mutlticast` | `underlay_mutlticast_pim_sm` or `<node_type_key>.defaults.underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].nodes[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.nodes[].underlay_multicast.pim_sm.enabled` |
| `underlay_ospf_authentication.message_digest_keys[].key` | `underlay_ospf_authentication.message_digest_keys[].cleartext_key` |

### eos_designs structured_configuration is now shared across eos_designs modules

> In AVD 5.x, the generation of strutcured configuration was done python module per python module.

Reviewer: ```suggestion
In AVD 5.x, the generation of structured configuration was done python module per python module.
```

> This implied that it was possible to declare conflicting information between modules (e.g. overwrite MLAG interface in connected_endpoint).

Reviewer: Not sure if using "modules" is appropriate from a user perspective. They don't really know which variables belong to which module.

Author: rewrite it as if you wanted a normal human to understand

Talk about duplication detection in data models

Author: changed - let me know :)

> This implies that it was possible to declare conflicting information between data models (e.g., overwriting an MLAG interface in `connected_endpoints`), which could lead to difficult-to-track errors.

Reviewer 2: ```suggestion
This implies that it was possible to declare conflicting information between data models (e.g. overwriting an MLAG interface in `connected_endpoints`), which could lead to difficult-to-track errors.
```

Author: with the comma is how it is supposed to be used in English

Author: https://www.scribbr.com/definitions/for-example-abbreviation/#:~:text=The%20latin%20abbreviation%20for%20%E2%80%9Cfor%20example%E2%80%9D%20is%20e.g.%2C%20which,)%20are%20native%20to%20Australia.%E2%80%9D

> Starting with AVD 6.0.0, conflict detection is now done across all data models, allowing for much stronger detection of these conflicting requirements.

Reviewer 2: ```suggestion
Starting with AVD 6.0.0, conflict detection is now done across all data models, allowing for much stronger detection of these conflicting configurations.
```

> You may experience errors when migrating. In general, these errors indicate that you had conflicting inputs across different data models in your variables and you should adjust them accordingly.

Reviewer 2: We should add an example of such an error so they know where to look.

Author: done

> You may experience errors when migrating, in general these errors would indicate that you had conflicting inputs across modules in your variables and you should adjust them.

Reviewer: Related to my other comment, maybe "configuration sections" or simply "sections" is more appropriate?
```suggestion
You may experience errors when migrating, but in general, these errors indicate that you had conflicting inputs across different configuration sections in your variables and you should adjust them accordingly.
```

Starting with AVD 6.0.0, the default value of  `underlay_ospf_graceful_restart` has been changed to `true`.

> to retain the previous behavior, the following change is required:

Author: ```suggestion
To retain the previous behavior, the following change is required:
```
